### ISSUE TITLE
[gql_dio_link] Converting object to an encodable object failed: Instance of 'MultipartFile'

### ISSUE THREAD

**Issue by kateile** [gql-dart/gql#259]
When I try to upload file via **gql_dio_link** I get the following error 

> Converting object to an encodable object failed: Instance of 'MultipartFile'

I am using the latest alpha version which support file uploads after this [PR ](https://github.com/gql-dart/gql/pull/255) made by @TarekkMA 

Demo project showing this problem is [here](https://github.com/kateile/gql_dio_link_upload_tester)

Any hints for solving this will be appreciated...

**Comment by TarekkMA** [comment]
I tried to run your project and the `MultipartFile` got encoded correctly.

BTW there was a bug in the code, you didn't await for `MultipartFile.fromFile(....)`

```dart
variables: {
  "input": await MultipartFile.fromFile(
    file.path,
    filename: "image.png",
    contentType: MediaType("image", "png"),
  )
},
```

**Comment by kateile** [comment]
Actually this bug is found in `graphql_flutter` and not `gql_dio_link`. It seems like graphql_flutter is doing some caching to MultipartFile and fail then it throws exception. I think same thing happens in http.MultipartFile  as well as dio.MultipartFile.

I used hint I found [here ](https://github.com/zino-app/graphql-flutter/issues/723) for solving this. 
Just add `fetchPolicy: FetchPolicy.noCache` for your mutation if you face same thing..

@TarekkMA Thanks for help. **gql_dio_link** is really amazing.
